python/pyspark/sql/pandas/group_ops.py

```diff
@@ -74,6 +79,11 @@ def apply(self, udf):
 
         assert isinstance(self, GroupedData)
 
+        warnings.warn(
+            "It is preferred to use 'applyInPandas' over this "
+            "API. This API will be deprecated in the future releases. See SPARK-28264 for "
+            "more details.", UserWarning)
```

Comment:
Seems trying to always show this warning even when using `applyInPandas`?
Maybe we should move the following logic into `applyInPandas` and call it from here after showing the warning?